presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSinkProvider.java

```diff
@@ -65,6 +65,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
         checkState(memoryInsertTableHandle.getActiveTableIds().contains(tableId));
 
         pagesStore.cleanUp(memoryInsertTableHandle.getActiveTableIds());
+        pagesStore.initialize(tableId);
```

Comment:
This seems to be a bug. Please consider following scenario, there had been some write query and so worker stored data for this table, then there was a crash and worker got restarted. Now, with your change, when a new insert comes, table will be initialized again and nobody would notice the data loss.

I wish we had tests for that.